src/apps/activities/crud/activity.py

```diff
@@ -103,3 +106,78 @@ async def get_ids_by_applet_id(self, applet_id: uuid.UUID) -> list[uuid.UUID]:
         query = query.where(ActivitySchema.applet_id == applet_id)
         result = await self._execute(query)
         return result.scalars().all()
+
+    async def get_activity_and_flow_basic_info_by_ids_or_auto(
+        self, applet_id: uuid.UUID, ids: list[uuid.UUID], language: str
+    ) -> list[ActivityOrFlowBasicInfoInternal]:
+        activities_query: Query = select(
+            ActivitySchema.id,
+            ActivitySchema.name,
+            ActivitySchema.description,
+            ActivitySchema.image.label("images"),
+            literal("").label("activity_ids"),
+            literal(False).label("is_flow"),
+            ActivitySchema.auto_assign,
+            ActivitySchema.is_hidden,
+            ActivitySchema.is_performance_task,
+            ActivitySchema.performance_task_type,
+        ).where(
+            ActivitySchema.applet_id == applet_id,
+            or_(
+                ActivitySchema.id.in_(ids),
+                ActivitySchema.auto_assign.is_(True),
+            ),
+        )
+
+        flow_alias = aliased(ActivityFlowSchema)
+        flow_items_alias = aliased(ActivityFlowItemSchema)
+        activities_alias = aliased(ActivitySchema)
+
+        flows_query = (
+            select(
+                flow_alias.id,
+                flow_alias.name,
+                flow_alias.description,
+                func.coalesce(
+                    func.string_agg(activities_alias.image, ",").filter(
+                        and_(activities_alias.image.isnot(None), activities_alias.image != "")
+                    ),
+                    "",
+                ).label("images"),
+                func.string_agg(activities_alias.id.cast(String), ",").label("activity_ids"),
+                literal(True).label("is_flow"),
+                flow_alias.auto_assign,
+                flow_alias.is_hidden,
+                literal(None).label("is_performance_task"),
+                literal(None).label("performance_task_type"),
+            )
+            .join(flow_items_alias, flow_alias.id == flow_items_alias.activity_flow_id)
+            .join(activities_alias, flow_items_alias.activity_id == activities_alias.id)
+            .where(
+                flow_alias.applet_id == applet_id,
+                or_(
+                    flow_alias.id.in_(ids),
+                    flow_alias.auto_assign.is_(True),
+                ),
+            )
+            .group_by(flow_alias.id, flow_alias.name, flow_alias.description, flow_alias.auto_assign)
+        )
+
+        union_query = union(activities_query, flows_query).order_by(text("is_flow DESC"))
```

Comment:
I know there's been some debate as to how activities/flows are ordered, but I believe we settled on aligning to the status quo for now – listing flows before activities (as you've done here), but then secondary sorting should be by activity/flow `order` as defined in the builder, which is missing. Could you add that here? 🙏🏻